common/automl.py

```diff
@@ -1,224 +1,32 @@
-        predictions = model.predict(x_test)
-        model_accuracy = roc_auc_score(y_test, predictions)
-
-        self.score = model_accuracy
-        self.model = model
-
-    def compare_score(self, score: float) -> bool:
-        return self.score > score
-
-
-class ScikitLearnRidgeClassifier(AutoMLModel):
-    def __init__(self) -> None:
-        super().__init__("Scikit-Learn RidgeClassifier", AutoMLModel.CLASSIFIER)
-
-    def train(
-        self,
-        x_train: pd.DataFrame,
-        y_train: pd.DataFrame,
-        x_test: pd.DataFrame,
-        y_test: pd.DataFrame,
-        x_val: pd.DataFrame,
-        y_val: pd.DataFrame,
-        features: List[str],
-        target: str,
-    ) -> None:
-        model = RidgeClassifier()
-        model.fit(x_train, y_train)
-        predictions = model.predict(x_test)
-        model_accuracy = roc_auc_score(y_test, predictions)
-
-        self.score = model_accuracy
-        self.model = model
-
-    def compare_score(self, score: float) -> bool:
-        return self.score > score
-
-
-class ScikitLearnDecisionTreeClassifier(AutoMLModel):
-    def __init__(self) -> None:
-        super().__init__("Scikit-Learn DecisionTreeClassifier", AutoMLModel.CLASSIFIER)
-
-    def train(
-        self,
-        x_train: pd.DataFrame,
-        y_train: pd.DataFrame,
-        x_test: pd.DataFrame,
-        y_test: pd.DataFrame,
-        x_val: pd.DataFrame,
-        y_val: pd.DataFrame,
-        features: List[str],
-        target: str,
-    ) -> None:
-        model = DecisionTreeClassifier(max_depth=5)
-        model.fit(x_train, y_train)
-        predictions = model.predict(x_test)
-        model_accuracy = roc_auc_score(y_test, predictions)
-
-        self.score = model_accuracy
-        self.model = model
-
-    def compare_score(self, score: float) -> bool:
-        return self.score > score
-
-
-class ScikitLearnAdaBoostClassifier(AutoMLModel):
-    def __init__(self) -> None:
-        super().__init__("Scikit-Learn AdaBoostClassifier", AutoMLModel.CLASSIFIER)
-
-    def train(
-        self,
-        x_train: pd.DataFrame,
-        y_train: pd.DataFrame,
-        x_test: pd.DataFrame,
-        y_test: pd.DataFrame,
-        x_val: pd.DataFrame,
-        y_val: pd.DataFrame,
-        features: List[str],
-        target: str,
-    ) -> None:
-        model = AdaBoostClassifier()
-        model.fit(x_train, y_train)
-        predictions = model.predict(x_test)
-        model_accuracy = roc_auc_score(y_test, predictions)
-
-        self.score = model_accuracy
-        self.model = model
-
-    def compare_score(self, score: float) -> bool:
-        return self.score > score
-
-
 class AutoML:
-    automl_models = [
-        ScikitLearnRandomForestClassifier(),
-        ScikitLearnDecisionTreeClassifier(),
-        ScikitLearnAdaBoostClassifier(),
-        ScikitLearnRidgeClassifier(),
-        ScikitLearnLinearRegression(),
-        ScikitLearnDecisionTreeRegressor(),
+    automl_models: Any = [
```

Comment:
```suggestion
    automl_models: Type[AutoMLModel] = [
```

Reply:
I tried it with 
 ```
automl_models: List[Type[AutoMLModel]]
```

but then i get this error. 

```
$ poetry run mypy .
common/automl.py:57: error: Missing positional arguments "name", "model_type" in call to "AutoMLModel"
```

This is because of I set the init parameters of model inside the subclasses. Couldn't find a way to fix it.

```python
class ScikitLearnRandomForestClassifier(AutoMLModel):
    def __init__(self) -> None:
        super().__init__("Scikit-Learn RandomForestClassifier", AutoMLModel.CLASSIFIER)
```